hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HStore.java

```diff
@@ -1137,7 +1146,13 @@ public StoreFileWriter createWriterInTmp(long maxKeyCount, Compression.Algorithm
           cacheOnWriteLogged = true;
         }
       } else {
-        writerCacheConf.setCacheDataOnWrite(false);
+        writerCacheConf.disableCacheOnWrite();
+        if (totalCompactedFileSize > cacheConf.getCacheCompactedBlocksOnWriteThreshold()) {
```

Comment:
Good.
may be we can log this line if the conf says to cache it but size is more. I would say use INFO log then.
Also to create the msg use the log statement like above way
LOG.info("For Store {} , cacheCompactedBlocksOnWrite....